scope_registry.go

```diff
@@ -138,15 +169,15 @@ func (r *scopeRegistry) Subscope(parent *scope, prefix string, tags map[string]s
 		bucketCache:     parent.bucketCache,
 		done:            make(chan struct{}),
 	}
```

Comment:
actually, it looks like this could be done outside of lock completely, or at least partially - so all the allocs wouldn't be done under lock - this should yield another perf improvement as lock would be held for a shorter time

Reply:
If we do this, there is a potential case where many goroutines are trying to acquire the write lock for the same key, so the allocations would be done for all of them.

Reply:
honestly, re-reading this method, I don't even know why I thought this could be done outside of lock ...